sound/soc/sof/pcm.c

```diff
@@ -619,6 +619,30 @@ static int sof_pcm_new(struct snd_soc_component *component,
 	return 0;
 }
 
+static void ssp_dai_config_pcm_params_match(struct snd_sof_dev *sdev, const char *link_name,
+					    struct snd_pcm_hw_params *params)
+{
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_dai *dai;
+	int i;
+
+	/*
+	 * Search for all matching DAIs as we can have both playback and capture DAI
+	 * associated with the same link.
+	 */
+	list_for_each_entry(dai, &sdev->dai_list, list) {
+		if (!dai->name || strcmp(link_name, dai->name))
+			continue;
+		for (i = 0; i < dai->number_configs; i++) {
+			config = &dai->dai_config[i];
+			if (config->ssp.fsync_rate == params_rate(params)) {
+				dev_dbg(sdev->dev, "DAI config %d matches pcm hw params\n", i);
+				dai->current_config = i;
```

Comment:
you don't want to break the inner loop here?

Reply:
no, I have to go through all DAI's associated with the same link.

Reply:
So, you want to find the last config which matches?
The loop can be written this way to achieve the same result:
for (i = dai->number_configs; i; i--) {
	config = &dai->dai_config[i - 1];
	if (config->ssp.fsync_rate == params_rate(params)) {
		dev_dbg(sdev->dev, "DAI config %d matches pcm hw params\n", i);
		dai->current_config = i - 1;
		break;
	}
}

It is true that you would not have dev_dbg() prints for all matches, but if this is the intention I would update the comment to reflect it.

Reply:
no I think the idea is to find the first matching config but set the config for all matching DAIs. So the loop looks good to me